tests/test_grouping.py

```diff
@@ -0,0 +1,341 @@
+        symplectic vector space."""
+
+        p1_vec = np.array([1, 0, 1, 0, 1, 1, 0, 0])
+        p2_vec = np.array([0, 0, 1, 1, 1, 0, 1, 0])
+        p3_vec = np.array([1, 1, 0, 1, 1, 1, 0, 1])
+
+        assert binary_symplectic_inner_prod(p1_vec, p2_vec) == 0
+        assert binary_symplectic_inner_prod(p1_vec, p3_vec) == 1
+        assert binary_symplectic_inner_prod(p2_vec, p3_vec) == 0
+
+    def test_is_pauli_word(self):
+        """Test for determining whether input `Observable` instance is a Pauli word."""
+
+        observable_1 = PauliX(0)
+        observable_2 = PauliZ(1) @ PauliX(2) @ PauliZ(4)
+        observable_3 = PauliX(1) @ Hadamard(4)
+        observable_4 = Hadamard(0)
+
+        assert is_pauli_word(observable_1)
+        assert is_pauli_word(observable_2)
+        assert not is_pauli_word(observable_3)
+        assert not is_pauli_word(observable_4)
+
+    def test_are_identical_pauli_words(self):
+        """Tests for determining if two Pauli words have the same `wires` and `name` attributes."""
+
+        pauli_word_1 = PauliX(0) @ PauliY(1)
+        pauli_word_2 = PauliY(1) @ PauliX(0)
+        pauli_word_3 = Tensor(PauliX(0), PauliY(1))
+        pauli_word_4 = PauliX(1) @ PauliZ(2)
+
+        assert are_identical_pauli_words(pauli_word_1, pauli_word_2)
+        assert are_identical_pauli_words(pauli_word_1, pauli_word_3)
+        assert not are_identical_pauli_words(pauli_word_1, pauli_word_4)
+        assert not are_identical_pauli_words(pauli_word_3, pauli_word_4)
+
+
+class TestPauliGroupingStrategy:
+    """Tests for the PauliGroupingStrategy class"""
+
+    def test_construct_complement_adj_matrix_for_operators(self):
+        """Constructing the complement graph adjacency matrix for a list of Pauli words and a
+        given symmetric binary relation."""
+
+        observables = [PauliY(0), PauliZ(0) @ PauliZ(1), PauliY(0) @ PauliX(1)]
+
+        qwc_complement_adjacency_matrix = np.array([[0, 1, 0], [1, 0, 1], [0, 1, 0]])
+
+        commuting_complement_adjacency_matrix = np.array([[0, 1, 0], [1, 0, 0], [0, 0, 0]])
+
+        anticommuting_complement_adjacency_matrix = np.array([[0, 0, 1], [0, 0, 1], [1, 1, 0]])
+
+        grouping_instance = PauliGroupingStrategy(observables, "qwc")
+        assert (
+            grouping_instance.construct_complement_adj_matrix_for_operator()
+            == qwc_complement_adjacency_matrix
+        ).all()
+        grouping_instance = PauliGroupingStrategy(observables, "commuting")
+        assert (
+            grouping_instance.construct_complement_adj_matrix_for_operator()
+            == commuting_complement_adjacency_matrix
+        ).all()
+        grouping_instance = PauliGroupingStrategy(observables, "anticommuting")
+        assert (
+            grouping_instance.construct_complement_adj_matrix_for_operator()
+            == anticommuting_complement_adjacency_matrix
+        ).all()
```

Comment:
Might be worth breaking up this test into three separate ones for readability: qwc, commuting and anticommuting